src/transport/connection.rs

```diff
@@ -141,6 +141,37 @@ impl BoundPort {
         initial: T,
         f: F)
         -> impl Future<Item = (), Error = io::Error> + Send + 'static
+        where
+            F: Fn(T, (Connection, SocketAddr)) -> Fut + Send + 'static,
+            T: Send + 'static,
+            Fut: IntoFuture<Item = T, Error = std::io::Error> + Send + 'static,
+            <Fut as IntoFuture>::Future: Send,
+    {
+        self.listen_and_fold_(std::u64::MAX, initial, f)
```

Comment:
I'm slightly uncomfortable with the use of `u64::MAX` as a connection limit. I don't really expect we'll hit  2E64 - 1 connections in practice, but I think passing the limit as an `Option` might be better...

Reply:
I _think_ something like

```rust
let stream = match connection_limit {
    Some(limit) => stream::Either::A(listener.incoming().take(limit)),
    None => stream::Either::B(listener.incoming()),
};
stream.and_then(...
```
ought to work?

Though you're right that we're probably not going to hit that limit.

Reply:
> I think something like

I spent some time trying it. Maybe there is some way to get it to work but I don't think it would be better than just using `usize::MAX` as the limit considering the extra complexity (in fact i didn't get something that can even build yet). I think we should just go with the original way.

Reply:
Fair enough!

Reply:
OK, I figured out what I was doing wrong: Using the wrong `Either`. Now I have it working both ways, but I still think the original way is clearer so I'm going to keep it that way.